app/modules/session/session.server.ts

```diff
@@ -38,3 +45,27 @@ export async function getUserSession(request: Request): Promise<UserSession | nu
   }
   return null;
 }
+
+export async function getCurrentUser(request: Request) {
+  const session = await getUserSession(request);
+
+  if (session) {
```

Comment:
Tiny nit (don't feel the need to update this): You could avoid some nesting by switching the if-condition around.

```
if(!session) {
  return null;
}

const currentUser = ...
```

By returning the edge cases early, you avoid nesting the main code logic deeper and deeper, which may improve readability. 

Totally fine here, just a nice pattern if a function grows in size.

Reply:
Noted. I'll keep this in mind for the future. Thanks for the tip!